### PR TITLE
Add removeActivityCost and doubleConfirm methods

### LINKED ISSUE #127: add removeActivityCost method to Person
removeActivityCost(int activityId) removes the ActivityCost with the matching activityId from the list of ActivityCosts associated with this Person.

### LINKED ISSUE #128: Add doubleConfirm method to TextUI
doubleConfirm(String message) prints message to the UI and asks the user for confirmation.
It should return true if the user confirms and false if the user does not.

### PR DESCRIPTION
**Pull request by matheril**
Person:
Added removeActivityCost(int activityId) method that removes the ActivityCost object with the corresponding activityId from this Person object.

ActivityCost:
Added equals and hashCode override methods. ActivityCost objects now check for equality using their activityIds. Two ActivityCost objects with the same ActivityId will be considered equal by the equals method.

TextUI:
Added doubleConfirm(String message) method that prints the message to the UI and asks for user confirmation.
Inputting Y/y/yes to the interface returns true.
Inputting N/n/no to the interface returns false and prints a message indicating that the user did not confirm.
Any other input prints an error message and asks the user to re-input.

Fixes #127
Fixes #128

### CODE REVIEW

File: src/main/java/seedu/splitlah/data/ActivityCost.java [hunk 1]
```diff
@@ -68,4 +70,20 @@ public void setCostOwed(int costOwed) {
...
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ActivityCost that = (ActivityCost) o;
```
Comment (froststein) [inline src/main/java/seedu/splitlah/data/ActivityCost.java:81]: Consider renaming the variable `that` to give more information

File: src/main/java/seedu/splitlah/ui/TextUI.java [hunk 2]
```diff
@@ -32,6 +32,34 @@ public String readNextLine() {
...
 
+    /**
+     * Requests confirmation from the user via the user interface.
+     * The user may confirm by inputting Y/y/yes to the interface.
+     * The user may reject by inputting N/n/no to the interface.
+     *
+     * @param message A message to be printed initially to the user interface.
+     * @return true if the user confirms
+     */
+    public boolean doubleConfirm(String message) {
```
Comment (froststein) [inline src/main/java/seedu/splitlah/ui/TextUI.java:43]: Perhaps rename this function to `getUserConfirmation`

**Review by froststein — CHANGES_REQUESTED** [review]
Some changes to be made for naming conventions.

File: src/main/java/seedu/splitlah/data/ActivityCost.java (hunk 1 above)
Comment (matheril) [reply src/main/java/seedu/splitlah/data/ActivityCost.java]: Resolved by removing this method

File: src/main/java/seedu/splitlah/ui/TextUI.java (hunk 2 above)
Comment (matheril) [reply src/main/java/seedu/splitlah/ui/TextUI.java]: Resolved

File: src/main/java/seedu/splitlah/ui/TextUI.java [hunk 3]
```diff
@@ -32,6 +32,35 @@ public String readNextLine() {
...
+        printDivider();
+        printlnMessage(message);
+        printlnMessage(Message.PROMPT_TEXTUI_REQUEST_CONFIRMATION);
+        String confirmation = readNextLine().toLowerCase();
+        printDivider();
+        while (true) {
+            switch (confirmation) {
+            case ("yes"):
+            case ("y"):
+                return true;
```
Comment (warrencxw) [inline src/main/java/seedu/splitlah/ui/TextUI.java:53]: Consider adding fallthrough comments here to make it clear that there is a fallthrough.

**Review by froststein — APPROVED** [review]
Looks good to merge

**Review by warrencxw — CHANGES_REQUESTED** [review]
Would be good to add fallthrough comments in switch-case block.

File: src/main/java/seedu/splitlah/ui/TextUI.java (hunk 3 above)
Comment (matheril) [reply src/main/java/seedu/splitlah/ui/TextUI.java:53]: Resolved

**Review by warrencxw — APPROVED** [review]
Looks good to merge.

**Pull request merged**